store/sqlstore/channel_store.go

```diff
@@ -1814,6 +1815,70 @@ func (s SqlChannelStore) UpdateLastViewedAt(channelIds []string, userId string)
 	return times, nil
 }
 
+// CountUnreadSince gives the number of posts that a channel has since some date, this doesn't follow any threads.
+// this allows us to know how many posts have bean read already.
+func (s SqlChannelStore) CountUnreadSince(channelID string, since int64) (int64, *model.AppError) {
```

Comment:
A bit more nitpicking about naming, but maybe this should be called `CountPostsSince` instead of `CountUnreadsSince` because it doesn't care whether or not the posts have been read or not.